frontends/common/parser_options.cpp

```diff
@@ -485,18 +487,15 @@ void ParserOptions::dumpPass(const char *manager, unsigned seq, const char *pass
             exit(1);
         }
         if (match) {
-            // FIXME: switch to sane printing via abseil
-            char buf[16];
-            snprintf(buf, sizeof(buf), "-%04zu-", ++dump_uid);
-            cstring suffix = cstring(buf) + name;
-            cstring filename = file;
-            if (filename == "-") filename = "tmp.p4"_cs;
-
-            cstring fileName = makeFileName(dumpFolder, filename, suffix);
+            std::string suffix = absl::StrFormat("-%04zu-%s", ++dump_uid, name);
```

Comment:
Can `absl::StrFormat` be replaced by `std::format` when we move to C++20?

Reply:
Yes, that's the intention. They are not quite compatible, but good thing is that `StrFormat` uses printf-style POSIX format strings (so, no new syntax as with boost::format), plus have compile-time checking of them.